cfripper/model/utils.py

```diff
@@ -47,7 +47,8 @@ def extract_bucket_name_and_path_from_url(url):
         bucket_name, path = match.group(1), match.group(2)[1:]  # Trim start /
 
     # https://bucket.s3-aws-region.amazonaws.com/path1/path2
-    match = re.search(r"^https://([^.]+).s3-[^\.]+.amazonaws.com(.*?)$", url)
+    # https://bucket.s3.aws-region.amazonaws.com/path1/path2
+    match = re.search(r"^https://([^.]+).s3[-\.]\b[^\.]+.amazonaws.com(.*?)$", url)
```

Comment:
`s3-[^\.]+` is replaced by `s3[-\.]\b[^\.]+`
Could you please explain what is the `\b`for?

Reply:
It prevents "s3--region" being valid and forces either "s3.region" or "s3-region".

Reply:
Thanks for clarifying this :). Smart solution.
Do you think adding some tests for the regex matching would be worth it? I'm thinking to something similar to this:
https://github.com/Skyscanner/pycfmodel/blob/master/tests/test_regexs.py

Reply:
(I don't normally use python or pytest but looked at this as an exercise)

So, along the lines of below incomplete example (2 & 3 pass, 1 & 4 fail)?

`from pycfmodel.constants import CONTAINS_STAR, CONTAINS_CF_PARAM, CONTAINS_S3_URL`

`@pytest.mark.parametrize(
    "string, expected",
    [
        ("https://bucket.s3.amazonaws.com/path1/path2", [('bucket', '/path1/path2')]),
        ("https://bucket.s3-aws-region.amazonaws.com/path1/path2", [('bucket', '/path1/path2')]),
        ("https://bucket.s3.aws-region.amazonaws.com/path1/path2", [('bucket', '/path1/path2')]),
        ("https://s3.amazonaws.com/bucket/path1/path2", [('bucket', '/path1/path2')]),
    ],
)`

`def test_contains_s3_url(string, expected):
    assert CONTAINS_S3_URL.findall(string) == expected`

and

`CONTAINS_S3_URL = re.compile(r"^https://([^.]+).s3[-\.]\b[^\.]+.amazonaws.com(.*?)$")`

Reply:
Something like that would work, yes.